tools/SwiftSyntax/SyntaxData.swift

```diff
@@ -38,6 +38,79 @@ final class SyntaxData: Equatable {
 
   let childCaches: [AtomicCache<SyntaxData>]
 
+  let positionCache: AtomicCache<AbsolutePosition>
+
+  let positionAfterSkippingLeadingTriviaCache: AtomicCache<AbsolutePosition>
+
+  let byteSizeCache: AtomicCache<ByteSize>
```

Comment:
Similarly, `byteSize` can be derived from the difference between `self.position.offset` and the `position.offset` of the next sibling. (If `self` is the last sibling in the node, we can use the next sibling of the *parent* node.)

Reply:
yeah, I think sharing cache is a good idea. I've changed the implementation to the way you suggest.